webapp/javascript/components/TimelineChartWrapper.jsx

```diff
@@ -157,10 +157,28 @@ class TimelineChartWrapper extends React.Component {
   };
 
   render = () => {
+    // Since profiling data is chuked by 10 seconds slices
+    // it's more user friendly to point a `center` of a data chunk
+    // as a bar rather than starting point, so we add 5 seconds to each chunk to 'center' it
     const flotData = this.props.timeline
-      ? [this.props.timeline.map((x) => [x[0], x[1] === 0 ? null : x[1] - 1])]
+      ? [
+          this.props.timeline.map((x) => [
+            x[0] + 5000,
+            x[1] === 0 ? null : x[1] - 1,
+          ]),
+        ]
       : [];
 
+    // In case there are few chunks left, then we'd like to add some margins to
+    // both sides making it look more centers
+    const flotOptions = {
```

Comment:
You're not actually using this one later on line :187


Other than that it looks and works great!